### PR TITLE
lxd: drop simplestreams dependency (SC-1437)

### PR DESCRIPTION
**Pull request by aciba90**
## Proposed Commit Message

```
lxd: drop simplestreams dependency

Refactor lxd images' location to use lxc's cli instead of
simplestreams.

This change makes pycloudlib easier to distribute and
fixing a logging config problem associated with simplestreams.
```

## Additional Context

https://github.com/canonical/pycloudlib/issues/267
https://github.com/canonical/pycloudlib/issues/257

### CODE REVIEW

File: pycloudlib/lxd/cloud.py [hunk 1]
```diff
@@ -21,6 +20,10 @@ class _BaseLXD(BaseCloud):
     _releases_remote = "ubuntu"
     _lxd_instance_cls = LXDInstance
 
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self._is_container = True
```
Comment (TheRealFalcon) [inline pycloudlib/lxd/cloud.py:25]: Does it make more sense to make this a class variable with just an annotation but no initialization? Then the child classes can override the value without needing a default here.

File: pycloudlib/lxd/_images.py [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+            remote = cls._REMOTE_DAILY
+        if ":" not in remote:
+            remote += ":"
+        return remote
+
+    @classmethod
+    def _find_images(
+        cls, remote: str, filters: Optional[Sequence[Tuple[str, str]]] = None
+    ) -> List[dict]:
+        """Find the latest image for a given release.
```
Comment (TheRealFalcon) [inline pycloudlib/lxd/_images.py:133]: docstring out of date

File: pycloudlib/lxd/_images.py [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+
+    @classmethod
+    def find_last_fingerprint(
+        cls,
+        daily: bool,
+        release: str,
+        is_container: bool,
+        arch: str,
+    ) -> Optional[str]:
+        """Find the LXD fingerprint in a given remote.
```
Comment (TheRealFalcon) [inline pycloudlib/lxd/_images.py:24]: docstring is out of date

**Review by TheRealFalcon — CHANGES_REQUESTED** [review]
Overall looks really good and nice refactor. I left a few comments, but nothing major.

File: pycloudlib/lxd/cloud.py (hunk 1 above)
Comment (aciba90) [reply pycloudlib/lxd/cloud.py]: I think it makes more sense to declare `_BaseLXD` as an abstract class and type-annotate `_is_container` within it. I have done that, let me know if you disagree.

**Comment by aciba90** [comment]
Thanks, @TheRealFalcon, for the review and suggestions. I believe I have addressed all of them.

**Review by TheRealFalcon — APPROVED** [review]
+1 but Travis tests are failing now. Once those pass we should be good to go.

**Comment by aciba90** [comment]
Close to re-trigger CI.

**Pull request merged**